src/sentry/tasks/reports.py

```diff
@@ -238,6 +241,94 @@ def prepare_project_report(interval, project):
     )
 
 
+class ReportBackend(object):
+    def build(self, timestamp, duration, project):
+        return prepare_project_report(
+            _to_interval(timestamp, duration),
+            project,
+        )
+
+    def prepare(self, timestamp, duration, organization):
+        """
+        Build and store reports for all projects in the organization.
+        """
+        raise NotImplementedError
+
+    def fetch(self, timestamp, duration, organization, projects):
+        """
+        Fetch reports for a set of projects in the organization, returning
+        reports in the order that they were requested.
+        """
+        raise NotImplementedError
+
+
+class DummyReportBackend(ReportBackend):
+    def prepare(self, timestamp, duration, organization):
+        pass
+
+    def fetch(self, timestamp, duration, organization, projects):
+        assert all(project.organization_id == organization.id for project in projects)
+        return map(
+            functools.partial(
+                self.build,
+                timestamp,
+                duration,
+            ),
+            projects,
+        )
+
+
+class RedisReportBackend(ReportBackend):
+    version = 0
+
+    def __init__(self, cluster, ttl, namespace='r'):
+        self.cluster = cluster
+        self.ttl = ttl
+        self.namespace = namespace
+
+    def __make_key(self, timestamp, duration, organization):
+        return '{}:{}:{}:{}:{}'.format(
+            self.namespace,
+            self.version,
+            organization.id,
+            int(timestamp),
+            int(duration),
+        )
+
+    def __encode(self, report):
+        return zlib.compress(json.dumps(report))
+
+    def __decode(self, value):
+        return json.loads(zlib.decompress(value))
+
+    def prepare(self, timestamp, duration, organization):
+        reports = {}
```

Comment:
Will this be problematic if people have multiple hundreds of projects? Definitely not a blocker, but the extreme side is ~1500 projects or so.